ratelimits/README.md

```diff
@@ -0,0 +1,181 @@
+# Configuring and Storing Key-Value Rate Limits
+
+## Rate Limit Structure
+
+All rate limits use a token-bucket model. The metaphor is that each limit is
+represented by a bucket which holds tokens. Each request removes some number of
+tokens from the bucket, or is denied if there aren't enough tokens to remove.
+Over time, new tokens are added to the bucket at a steady rate, until the bucket
+is full. The _burst_ parameter of a rate limit indicates the maximum capacity of
+a bucket: how many tokens can it hold before new ones stop being added.
+Therefore, this also indicates how many requests can be made in a single burst
+before a full bucket is completely emptied. The _count_ and _period_ parameters
+indicate the rate at which new tokens are added to a bucket: every period, count
+tokens will be added. Therefore, these also indicate the steady-state rate at
+which a client which has exhausted its quota can make requests: one token every
+(period / count) duration.
+
+## Default Limit Settings
+
+Each key directly corresponds to a `Name` enumeration as detailed in `name.go`.
+The Name enum is used to identify the particular limit. The parameters of a
+default limit are the values that will be used for all buckets that do not have
+an explicit override (see below).
+
+```yaml
+NewRegistrationsPerIPAddress:
+  burst: 20
+  count: 20
+  period: 1s
+NewOrdersPerAccount:
+  burst: 300
+  count: 300
+  period: 180m
+```
+
+## Override Limit Settings
+
+Each override key represents a specific bucket, consisting of two elements:
+_name_ and _id_. The name here refers to the Name of the particular limit, while
+the id is a client identifier. The format of the id is dependent on the limit.
+For example, the id for 'NewRegistrationsPerIPAddress' is a subscriber IP
+address, while the id for 'NewOrdersPerAccount' is the subscriber's registration
+ID.
+
+```yaml
+NewRegistrationsPerIPAddress:10.0.0.2:
+  burst: 20
+  count: 40
+  period: 1s
+NewOrdersPerAccount:12345678:
+  burst: 300
+  count: 600
+  period: 180m
+```
+
+The above example overrides the default limits for specific subscribers. They
+will be allowed to make twice as many requests as the default limits but will
+still be limited to the same burst as a regular subscriber.
```

Comment:
but are still configured with the default burst limit settings for the respective rate limit type.

Reply:
I almost read this as though it promises fall-through to the default for the rate-limit.

Reply:
Mulling this over for a while I went with:

> The above example overrides the default limits for specific subscribers. In both
cases the count of requests per period are doubled, but the burst capacity is
explicitly configured to match the default rate limit.

Thanks.